doc/grammars/type.abnf

```diff
@@ -60,12 +60,24 @@ Array
 	= 1*(TokenSquareBracketOpen TokenSquareBracketClose)
 
 ArrayShape
-    = TokenCurlyBracketOpen ArrayShapeItem *(TokenComma ArrayShapeItem) TokenCurlyBracketClose
+	= Shape
 
 ArrayShapeItem
+    = ShapeItem
+
+Shape
+    = TokenCurlyBracketOpen ShapeItem *(TokenComma ShapeItem) TokenCurlyBracketClose
+
+ShapeItem
     = (ConstantString / ConstantInt / TokenIdentifier) TokenNullable TokenColon Type
     / Type
 
+ObjectShape
+	= Shape
```

Comment:
nit: tabs vs spaces

Reply:
good spot

the `.editorconfig` doesn't contain an entry for `.abnf` files so i'll add that aswell